NMS-15425: Update dual write plugin docs (#5788)
OpenNMS/opennms: docs/modules/deployment/pages/time-series-storage/timeseries/cortex.adoc

. Download the plugin's `.kar` file from GitHub (https://github.com/OpenNMS/opennms-cortex-tss-plugin/releases).
. Copy the file into `$\{OPENNMS_HOME}/deploy`.

> . Enable and configure time series database plugins in `$\{OPENNMS_HOME}/etc/opennms.properties.d/timeseries.properties`:

Reviewer: There is a ticket, NMS-15410 that says `opennms.properties.d` doesn't exist under `/examples`. Freddy is investigating for me. What am I missing? :-)

Author: It exists for me on 31.0.4.

Reviewer: Ticket doesn't include "affects version" information. Probably don't need to worry about it here.

+
[source, properties]
----

> org.opennms.timeseries.strategy=integration <1><2>

Reviewer 2: ```suggestion
org.opennms.timeseries.strategy=integration <1> <2>
```

org.opennms.timeseries.tin.metatags.tag.node=${node:label} <3>
org.opennms.timeseries.tin.metatags.tag.location=${node:location}
org.opennms.timeseries.tin.metatags.tag.geohash=${node:geohash}
org.opennms.timeseries.tin.metatags.tag.ifDescr=${interface:if-description}
org.opennms.timeseries.tin.metatags.tag.label=${resource:label}
----

> <1> Make sure that the `org.opennms.timeseries.strategy` property is not set in `$\{OPENNMS_HOME}/etc/opennms.properties` or any other file in `$\{OPENNMS_HOME}/etc/opennms.properties.d/`.

Reviewer 3: This only applies for dual-writes. When *not* using dual-writes, this *must* be set.

Author: Intention here is to make sure the setting is only defined in one place. Need to clarify that.

Author: ```suggestion
<1> Make sure that the `org.opennms.timeseries.strategy` property is only set once so you don't have conflicting values in multiple `properties` files.
```

> <1> Make sure that the `org.opennms.timeseries.strategy` property is not set in `$\{OPENNMS_HOME}/etc/opennms.properties` or any other file within the `$\{OPENNMS_HOME}/etc/opennms.properties.d/` directory.

Reviewer 2: ```suggestion
<1> Make sure that the `org.opennms.timeseries.strategy` property is not set in `$\{OPENNMS_HOME}/etc/opennms.properties` or any other file in `$\{OPENNMS_HOME}/etc/opennms.properties.d/`.
```

> <2> If you are enabling the xref:time-series-storage/timeseries/time-series-storage.adoc#ga-dual-write-integration[dual write plugin] on an existing installation and want to keep historical metrics, do not set the `org.opennms.timeseries.strategy` to `integration` until you have written enough data to expire any collected data prior to enabling the dual write feature.

Reviewer 2: ```suggestion
<2> If you are enabling the <<time-series-storage/timeseries/time-series-storage.adoc#ga-dual-write-integration, dual write plugin>> on an existing installation and want to keep historical metrics, do not set the `org.opennms.timeseries.strategy` to `integration` until you have written enough data to expire any collected data prior to enabling the dual write feature.
```

org.opennms.timeseries.tin.metatags.tag.node=${node:label}<2>
org.opennms.timeseries.tin.metatags.tag.location=${node:location}
org.opennms.timeseries.tin.metatags.tag.geohash=${node:geohash}
org.opennms.timeseries.tin.metatags.tag.ifDescr=${interface:if-description}
org.opennms.timeseries.tin.metatags.tag.label=${resource:label}
----
<1> If you are enabling the xref:time-series-storage/timeseries/time-series-storage.adoc#ga-dual-write-integration[dual write plugin] on an existing installation and want to keep historical metrics, do not set the `org.opennms.timeseries.strategy` to `integration` until you have written enough data to expire any collected data prior to enabling the dual write feature.
<2> See the xref:time-series-storage/timeseries/configuration.adoc#ga-opennms-operation-timeseries-properties-meta-tags[time series integration] section for more info on defining metatags.
+
NOTE: If you are deploying to a container, instead of setting the `org.opennms.timeseries.strategy` via a properties file, update the environment variables of your container config:
+
[source, yaml]
    environment:
        OPENNMS_TIMESERIES_STRATEGY=integration
+

> IMPORTANT: Make sure that the `org.opennms.timeseries.strategy` property is not set in `$\{OPENNMS_HOME}/etc/opennms.properties` or any other file within the `$\{OPENNMS_HOME}/etc/opennms.properties.d/` directory.

Reviewer 2: Could this be included in the previous admonition to ensure that we're not using too many close together?

Author: Probably, though one of these definitely is more important than the other.
